luigi/contrib/k8s_job.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2015 Outlier Bio, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+"""
+Kubernetes Job wrapper for Luigi.
+
+From the Kubernetes website:
+
+    Kubernetes is an open-source system for automating deployment, scaling,
+    and management of containerized applications.
+
+For more information about Kubernetes Jobs: http://kubernetes.io/docs/user-guide/jobs/
+
+Requires:
+
+- pykube: ``pip install pykube``
+
+Written and maintained by Marco Capuccini (@mcapuccini).
+"""
+
+import luigi
+from luigi import configuration
+import logging
+import uuid
+import time
+
+logger = logging.getLogger('luigi-interface')
+
+try:
+    from pykube.config import KubeConfig
+    from pykube.http import HTTPClient
+    from pykube.objects import Job
+except ImportError:
+    logger.warning('pykube is not installed. KubernetesJobTask requires pykube.')
+
+
+class KubernetesJobTask(luigi.Task):
+
+    __POLL_TIME = 5  # see __track_job
+
+    def _init_k8s(self):
+        self.__logger = logger
+        self.__logger.debug("Kubernetes auth method: " + self.auth_method)
+        if(self.auth_method == "kubeconfig"):
+            self.__kube_api = HTTPClient(KubeConfig.from_file(self.kubeconfig_path))
+        elif(self.auth_method == "service-account"):
+            self.__kube_api = HTTPClient(KubeConfig.from_service_account())
+        else:
+            raise ValueError("Illegal auth_method")
+        self.job_uuid = str(uuid.uuid4().hex)
+        self.uu_name = self.name + "-luigi-" + self.job_uuid
+        if ("restartPolicy" not in self.spec_schema):
+            self.spec_schema["restartPolicy"] = "Never"
+
+    @property
+    def auth_method(self):
+        """
+        This can be set to ``kubeconfig`` or ``service-account``.
+        It defaults to ``kubeconfig``.
+
+        For more details, please refer to:
+
+        - kubeconfig: http://kubernetes.io/docs/user-guide/kubeconfig-file
+        - service-account: http://kubernetes.io/docs/user-guide/service-accounts
+        """
+        return configuration.get_config().get("k8s", "auth_method", "kubeconfig")
```

Comment:
At the top there's an import for `from luigi import configuration`. Would you prefer calling `luigi.configuration`?